### PR TITLE
Allow return anyway a solution even when running out of grib temporal scope

### PR DESCRIPTION
**Pull request by enricofer**
We (me and @pcav) noticed that when performing a routing, the algorithm does not find a solution inside the grib temporal scope, no output is returned. So I modify the router to stop in case it runs out of grib scope and to return anyway the best approaching solution found so far.
In this case we have to clearly define a convention: when probing the grib out of its temporal scope, the getWindAt class method must return `None` 
What do you think?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,9 +117,7 @@ the step method returns a RoutingResult object with the following informations d
 ```
 res.time         # the datetime of step  
 res.isochrones   # all points reached at a specified datetime
-res.progress     # the calculation progress
-res.path         # when the end track point is reached contains the route path
-res.position    
+res.progress     # the calculation progress
```
Comment (dakk) [inline README.md:120]: why did you remove path and position?

File: weatherrouting/routers/linearbestisorouter.py [hunk 2]
```diff
@@ -26,26 +26,43 @@ class LinearBestIsoRouter (Router):
 	}
 
 	def route (self, lastlog, time, start, end):
```
Comment (dakk) [inline weatherrouting/routers/linearbestisorouter.py:28]: I don't understand what you are doing here; can you please explain?

**Review by dakk — COMMENTED** [review]
It's ok for me to return None if no wind information are available for that space-time combination. Anyway, can you explain further what is doing route() now?

File: README.md (hunk 1 above)
Comment (enricofer) [reply README.md:120]: only because before routing.end path and position are void and become significant only when algorithm ends

File: weatherrouting/routers/linearbestisorouter.py [hunk 3]
```diff
@@ -26,26 +26,43 @@ class LinearBestIsoRouter (Router):
...
+				distance_to_end_point = utils.pointDistance (end[0],end[1], p[0], p[1])
+				if distance_to_end_point < self.getParamValue('minIncrease'):
+					(twd,tws) = self.grib.getWindAt (time + datetime.timedelta(hours=1), p[0], p[1])
+					maxReachDistance = self.polar.maxReachDistance(p, twd, tws)
+					if utils.pointDistance (end[0],end[1], p[0], p[1]) < abs(maxReachDistance*1.1):
+						if (not self.pointValidity or self.pointValidity(end[0],end[1])) and (not self.lineValidity or self.lineValidity(end[0],end[1], p[0], p[1])):
+							generate_path(p)
+							break
+						
+		else: #out of grib scope
```
Comment (enricofer) [inline weatherrouting/routers/linearbestisorouter.py:58]: The PR suggests a second exit condition: when algorithm goes out of grib temporal scope (getWindAt grib class method returns None) it rescans last calculated isochrones to find the nearest to end point of track and generate a partial solution with new generate_path function (created only to avoid code duplication...)

File: README.md (hunk 1 above)
Comment (dakk) [reply README.md:120]: That's not true; if the track has more than two points, path contains middle points even if the complete routing is not done.

File: weatherrouting/routers/linearbestisorouter.py (hunk 2 above)
Comment (dakk) [reply weatherrouting/routers/linearbestisorouter.py:28]: Ok; can you provide also a test which match this case?

File: weatherrouting/routers/linearbestisorouter.py (hunk 3 above)
Comment (dakk) [reply weatherrouting/routers/linearbestisorouter.py:58]: seems legit

File: weatherrouting/routers/linearbestisorouter.py (hunk 2 above)
Comment (enricofer) [reply weatherrouting/routers/linearbestisorouter.py:28]: https://github.com/dakk/libweatherrouting/pull/3/commits/e758b5fd3efe5ce67647103c9d690e49255ac7f5

**Review by dakk — APPROVED** [review]
